### PR TITLE
Try to make tooltip stay on the screen

### PR DESCRIPTION
**Pull request by SyDr**
This will work on all tooltips with this style (buildings, upgrades, achievements, santa upgrades).

### CODE REVIEW

File: src/CookieMonster/interface/tooltips.js [hunk 1]
```diff
@@ -152,6 +154,26 @@ CookieMonster.updateTooltips = function(which) {
...
+ * Try to make tooltip stay on the screen
+ *
+ * @return {void}
+ */
+CookieMonster.controlTooltipPosition = function() {
+	var yMax = Game.mouseY;
+
+	if (CookieMonster.tooltipLastObjectId) {
+		var elem1 = CookieMonster.tooltipLastObjectId + 'note_div_warning';
+		var elem2 = CookieMonster.tooltipLastObjectId + 'note_div_caution';
```
Comment (Anahkiasen) [inline src/CookieMonster/interface/tooltips.js]: Can you cache $(CookieMonster.tooltipLastObjectId + 'note_div_warning') as $elem1, same for $elem2 so the selectors are cached ?

File: src/CookieMonster/interface/tooltips.js [hunk 2]
```diff
@@ -152,6 +154,26 @@ CookieMonster.updateTooltips = function(which) {
...
+		var underLuckyHeight = ($(elem1).is(':visible') ? $(elem1).outerHeight(true): 0) + ($(elem2).is(':visible') ? $(elem2).outerHeight(true) + 2: 0); // 2 is a magic number
+		yMax = $(window).height() - ($(tooltip).outerHeight() + underLuckyHeight + 40 + (CookieMonster.getSetting('BottomBar') ? 57 : 0)); // 40 is not magic number
```
Comment (Anahkiasen) [inline src/CookieMonster/interface/tooltips.js]: Any way to split those two lines or add a comment, I usually restrict myself to 80 columns lines, not sure this goes overboard or not but it seems quite lengthy.

**Comment by Anahkiasen** [comment]
Added a few quick comments but overall really good job!

**Comment by Anahkiasen** [comment]
I think something got messed up here because you have a `tooltip` variable and a `tooltipAnchor` variable but neither of these two are declared anywhere ?

**Comment by SyDr** [comment]
As I say before: I don't know js (and jQuery), so I cannot create a good solution or find errors, if code works.
These variables are declared in Cookie Clicker itself. And $("tooltip") don't worked, but $(tooltip) works.

**Comment by Anahkiasen** [comment]
Merged this manually, thanks!

**Pull request closed without merging**